### PR TITLE
[Locked Figures] Use $ symbols to denote TeX within locked labels and locked figure labels

### PR DESCRIPTION
**Pull request by nishasy**
## Summary:
In the rest of the exercise editor, content authors need to surround text in `$` for it to be denoted as TeX.

For consistency, we want to add that functionality to the interactive graph editor as well.

- Update LockedLabelSettings to say "text" instead of "TeX" and have an infotip to explain it.
- Update LockedLabel so that it surrounds the non-$ part of the string with \text{} before
  wrapping the whole thing in <TeX>

Note: This is a prerequisite for auto-generating spoken math details with the MathJax speech engine
for the aria labels, because the non-TeX text needs to be differentiated in the label for the label
to be generated correctly for non-TeX text. If it's all TeX, the speech engine reads every letter
separately (example: "<TeX>square A</TeX>" ==> "s q u a r e upper A")

Issue: https://khanacademy.atlassian.net/browse/LEMS-2548

## Test plan:
`yarn jest packages/perseus/src/widgets/interactive-graphs/utils.test.ts`

Storybook
- http://localhost:6006/?path=/story/perseuseditor-widgets-interactive-graph--mafs-with-locked-figure-labels-all-flags
- Play around with the locked label and the locked figure labels

### CODE REVIEW

File: packages/perseus/src/widgets/interactive-graphs/utils.ts [hunk 1]
```diff
@@ -70,3 +70,27 @@ export function isUnlimitedGraphState(
...
+        if (currentlyTeX) {
+            result += part;
+        } else {
+            result += `\\text{${part}}`;
+        }
+        currentlyTeX = !currentlyTeX;
+    }
+
+    return result;
+}
```
Comment (nishasy) [inline packages/perseus/src/widgets/interactive-graphs/utils.ts:96]: I tried doing some regex stuff along with `mathString.replace()` but it got convoluted and kept letting edge cases slip through the cracks. I decided to go back to basics and stick to what works, and this was the resulting code.

File: packages/perseus/src/widgets/interactive-graphs/utils.test.ts [hunk 2]
```diff
@@ -65,3 +65,58 @@ describe("normalizeCoords", () => {
         expect(result).toEqual(expected);
     });
 });
+
+describe("replaceOutsideTeX", () => {
+    test("no $s", () => {
```
Comment (anakaren-rojas) [inline packages/perseus/src/widgets/interactive-graphs/utils.test.ts:70]: Since these tests do the same thing, using a table might make things easier since you won't need to redo the code for each test, just have a table of inputs
https://jestjs.io/docs/api#2-testeachtablename-fn-timeout

File: packages/perseus/src/widgets/interactive-graphs/utils.ts [hunk 3]
```diff
@@ -70,3 +70,27 @@ export function isUnlimitedGraphState(
         (state.type === "polygon" && state.numSides === "unlimited")
     );
 }
+
+/**
+ * Replace all text outside of the $ TeX blocks with `\\text{...}`
+ * This way, the entire resulting string can be rendered within <TeX>
+ * and the text outside of the $ blocks will be non-TeX text.
+ */
+export function replaceOutsideTeX(mathString) {
```
Comment (anakaren-rojas) [inline packages/perseus/src/widgets/interactive-graphs/utils.ts:79]: For type safety
```suggestion
export function replaceOutsideTeX(mathString: string) {
```

File: packages/perseus/src/widgets/interactive-graphs/utils.ts [hunk 4]
```diff
@@ -70,3 +70,27 @@ export function isUnlimitedGraphState(
...
+
+    const splitString = mathString.split("$").filter((part) => part !== "");
+
+    for (let i = 0; i < splitString.length; i++) {
+        const part = splitString[i];
+        if (currentlyTeX) {
+            result += part;
+        } else {
+            result += `\\text{${part}}`;
+        }
```
Comment (anakaren-rojas) [inline packages/perseus/src/widgets/interactive-graphs/utils.ts:91]:
```suggestion
result += currentlyTeX ? part : `\\text{${part}}`;
```

**Review by anakaren-rojas — APPROVED** [review]

File: packages/perseus/src/widgets/interactive-graphs/utils.test.ts (hunk 2 above)
Comment (nishasy) [reply packages/perseus/src/widgets/interactive-graphs/utils.test.ts:70]: I thought about that, but I wanted the description of each test to be explicit!

Comment (nishasy) [reply packages/perseus/src/widgets/interactive-graphs/utils.test.ts:70]: Also I think the table would get really long with some of the descriptions + math strings. It would definitely be fewer lines, but I think I prefer the deliberateness of having them be separate tests in this particular case.
